tests/test_tuya_smoke.py

```diff
@@ -1,68 +1,85 @@
-        "endpoints": {
-            "1": {
-                "profile_id": 0x0104,
-                "device_type": "0x0402",
-                "in_clusters": ["0x0000", "0x0001", "0x0004", "0x0005", "0x0500"],
-                "out_clusters": [
-                    "0x0003",
-                    "0x0004",
-                    "0x0006",
-                    "0x000a",
-                    "0x0019",
-                    "0x1000",
-                ],
-            }
-        },
-        "manufacturer": "_TZ3210_up3pngle",
-        "model": "TS0205",
-        "class": "ts0205.TuyaSmokeDetectorTS0205",
-    }
-    assert_signature_matches_quirk(
-        zhaquirks.tuya.ts0205.TuyaSmokeDetectorTS0205, signature
-    )
-
-
-@pytest.mark.parametrize("quirk", (zhaquirks.tuya.ts0205.TuyaSmokeDetectorTS0205,))
-async def test_tuya_smoke_sensor_attribute_update(zigpy_device_from_quirk, quirk):
-    """Test update_attribute on Tuya smoke sensor."""
-
-    device = zigpy_device_from_quirk(quirk)
-
-    opple_cluster = device.endpoints[1].tuya_manufacturer
-    opple_listener = ClusterListener(opple_cluster)
-
-    ias_cluster = device.endpoints[1].ias_zone
-    ias_listener = ClusterListener(ias_cluster)
+@pytest.mark.parametrize(
+    "model,manuf,battery_test",
+    [
+        ("_TZE200_dq1mfjug", "TS0601", []),
+        ("_TZE200_m9skfctm", "TS0601", []),
+        ("_TZE200_ntcy3xu1", "TS0601", []),
+        ("_TZE200_rccxox8p", "TS0601", []),
+        ("_TZE200_vzekyi4c", "TS0601", []),
+        ("_TZE204_vawy74yh", "TS0601", []),
+        (
+            "_TZE204_ntcy3xu1",
+            "TS0601",
+            (
+                (b"\x09\x3a\x02\x00\x12\x0e\x04\x00\x01\x02", 200),
+                (b"\x09\x3a\x02\x00\x12\x0e\x04\x00\x01\x01", 80),
+                (b"\x09\x3a\x02\x00\x12\x0e\x04\x00\x01\x00", 10),
+            ),
+        ),
+        ("_TZE284_0zaf1cr8", "TS0601", []),
+        ("_TZ3210_up3pngle", "TS0205", []),
+    ],
+)
+async def test_handle_get_data(zigpy_device_from_v2_quirk, model, manuf, battery_test):
+    """Test handle_get_data for multiple attributes."""
 
     zone_status_id = IasZone.AttributeDefs.zone_status.id
 
-    # check that updating smoke attribute also updates zone status on the Ias Zone cluster
+    quirked = zigpy_device_from_v2_quirk(model, manuf)
+    ep = quirked.endpoints[1]
+
+    assert ep.basic is not None
+    assert isinstance(ep.basic, Basic)
```

Comment:
Probably not really needed, as we're testing the quirk here and not that `zigpy_device_from_v2_quirk` returns a device with the `Basic` cluster, but it's fine.

Reply:
Agreed, removed